openedx/core/djangoapps/user_api/views.py

```diff
@@ -130,6 +136,25 @@ def post(self, request):
         email = data.get('email')
         username = data.get('username')
 
+        # Begin reCAPTCHA validation #
+        recaptcha_response = data.get('g-recaptcha-response')
+        url = settings.CAPTCHA_VERIFY_URL
+        values = {
+            'secret': settings.CAPTCHA_SECRET_KEY,
+            'response': recaptcha_response
+        }
+        data = urllib.urlencode(values)
+        req = urllib2.Request(url, data)
+        response = urllib2.urlopen(req)
+        result = json.load(response)
+        # End reCAPTCHA validation #
+
+        if not result['success']:
+            errors = {
+                'g_recaptcha': [{"user_message": "Invalid reCaptcha"}]
+            }
+            return JsonResponse(errors, status=400)
+
```

Comment:
This is a core file, in which we have tried to do very very minimal changes. Please have a look at [this](https://github.com/philanthropy-u/edx-platform/blob/72763424b62efabea3307cfb2a92f0b42faeab44/openedx/core/djangoapps/user_authn/views/register.py#L154). This is the form we have used for backend validations. 
I suggest you study how the validations are being done through this form, and make the captcha validation a part of it. Feel free to have a discussion on this any time.